File: 0981/Original-Submission/Src/ex3b_BVP/LTsings2.c
/**     EXAMPLE 3b

    Application of Talbot's method to solve
    the following PDE problem

        u_t (x,t) = u_xx (x,t),		0 < x < L, t > 0
		  u(x,0+) = x*(x-1)
		  u(0,t)  = 2*t
		  u(L,t)  = 2*t + L*(L-1)

    in [0,L]x[T0,T1].

    The analytical solution is

            u(x,t) = x*(x-1) + 2*t

    and Laplace Transform is

                     x*(x-1)    2
            U(x,s) = ------- + ---
                        s      s^2

    where s = 0 is a double pole.
*/

#include <complex.h>


unsigned int LTsings2(unsigned int *Nsings, double complex *SINGS, unsigned int *MULT, double *sigma0)
/* Abscissa of convergence and singularities of F(s) with their polar multiplicities */
{
    unsigned int NsingsTOT = 1; /* total number of singularities s_j */
    *Nsings = 1;                /* number of singularities s_j with Im(s_j) ge 0 */
    *sigma0 = 0;                /* abscissa of convergence */

    SINGS[0]=0.0+0.0*I;
    MULT[0]=2;

    return NsingsTOT;
}

File: 0981/Original-Submission/Src/ex5_PDE/LTsings2.m
function [NsingsTOT, Nsings, SINGS, MULT, sigma0] = LTsings2()
%{
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
                EXAMPLE 5

        Application of Talbot's method to solve
        the following PDE problem

                u_t  = u_xx  + u_yy,            0 < x,y < 1,   t>0
                u(x,y,0+) = x*(x-1) + y*(y-1)
                u(0,y,t) = u(1,y,t) = 4*t + y*(y-1)
                u(x,0,t) = u(x,1,t) = 4*t + x*(x-1)

        The analytical solution is

                u(x,y,t) = 4*t + x*(x-1) + y*(y-1)

        and Laplace Transform is

                          4    x*(x-1) + y*(y-1)
                U(x,s) = --- + -----------------
                         s^2           s

        where s = 0 is a double pole.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%}
    NsingsTOT = 1; % total number of singularities s_j
    Nsings = 1;    % number of singularities s_j with Im(s_j) ge 0
    sigma0 = 0;    % abscissa of convergence
    SINGS = 0;     % singularity
    MULT  = 2;     % polar multiplicity
end
